data-raw/england-aande.R

```diff
@@ -0,0 +1,91 @@
+  # get url from query_urls table
+  query_url <-
+    query_urls |>
+    filter(id == {{ id }}) |>
+    pull(query)
+
+  # download xls to as a temp file
+  download <- tempfile(fileext = ".xls")
+
+  request(query_url) |>
+    req_perform(download)
+
+  # read the xls file
+  raw <-
+    read_excel(
+      download,
+      sheet = sheet,
+      range = range
+    )
+
+  # clean the df
+  data <- raw |>
+    select(all_of(col_index)) |> # select relevant columns
+    rename(
+      code = 1, 
+      total_attendances = 2, 
+      attendances_over_4hours = 3, 
+      total_emergency_admissions = 4, 
+      emergency_admissions_over_4hours = 5
+    ) |> # rename columns
+    slice(-1) |> # remove first row
+    filter(!is.na(code)) |> # remove NAs
+    mutate_at(2:5, as.numeric) |> # change cols datatype to numeric
+    mutate(
+      pct_attendance_over_4hours = round(attendances_over_4hours / total_attendances, 2),
+      pct_emergency_admissions_over_4hours = round(emergency_admissions_over_4hours / total_emergency_admissions, 2),
+      date = date
+    ) # add percentage and date columns
+
+  return(data)
+}
+
+# ---- ICB level data ----
+# - Iterate over all data sets and return as a dataframe
+
+# Generate a dataframe with function arguments
+icb_df <-
+  tibble(
+    id = query_urls[which(query_urls$id == "nhs_accident_emergency_april_22"):(which(query_urls$id == "nhs_accident_emergency_april_22") + 12), ] |> pull(id),
+    sheet = rep("System Level Data", 13),
+    range = rep("B16:AA61", 13),
+    date = c("April 2022", "May 2022", "June 2022", "July 2022", "August 2022", "September 2022", "October 2022", "November 2022", "December 2022", "January 2023", "February 2023", "March 2023", "April 2023"),
+    col_index = rep(list(c(1, 6, 14, 24, 25)), 13)
+  )
+
+# Build dataframe with all months
+england_icb_accidents_emergency <- pmap_dfr(icb_df, scrape_data)
+
+# Save output to data/ folder
+usethis::use_data(england_icb_accidents_emergency, overwrite = TRUE)
+
+# ---- Provider level data ----
+# - Iterate over all data sets and return as a dataframe
+
+# Generate a dataframe with function arguments
+trust_df <-
+  tibble(
+    id = query_urls |> filter(str_detect(id, "^nhs_accident_emergency")) |> pull(id),
+    sheet = rep("Provider Level Data", 25),
+    range = rep("B16:AB223", 25),
+    date = c("April 2021", "May 2021", "June 2021", "July 2021", "August 2021", "September 2021", "October 2021", "November 2021", "December 2021", "January 2022", "February 2022", "March 2022", "April 2022", "May 2022", "June 2022", "July 2022", "August 2022", "September 2022", "October 2022", "November 2022", "December 2022", "January 2023", "February 2023", "March 2023", "April 2023"),
+    col_index = rep(list(c(1, 7, 15, 25, 26)), 25)
+  )
+
+# Build dataframe with all months
+england_trust_accidents_emergency <- pmap_dfr(trust_df, scrape_data)
+
+# Save output to data/ folder
+usethis::use_data(england_trust_accidents_emergency, overwrite = TRUE)
```

Comment:
There are appears to be lots of missing values in the trust data (download the `skimr` package to get this code to work):

```r
england_trust_accidents_emergency |>  
  filter(code %in% geographr::points_nhs_trusts22$nhs_trust22_code) |> 
  skimr::skim()
```
Looking at the raw data, i looks like lots of data is reported as `-`. Please check (through documentation and NHS site) whether this is due to missing data, or as a placeholder for a value of zero.

Reply:
From the statistical commentary my guess is that `-` are for providers that do not report on these indicators (because they are taking part in a trial) - which when processed becomes NaN. 

Zeros can represent trusts that did not submit data or perhaps real zeros.

I was going to note this in the documentation but doesn't look like any of others have these kinds of commentary. Let me know what you want me to do with this.

Reply:
https://www.england.nhs.uk/statistics/wp-content/uploads/sites/2/2023/04/Statistical-commentary-March-2023-assefg.pdf

Reply:
Added notes to the documentation